// definebabe/dist/lib/gallery.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _promise = require('babel-runtime/core-js/promise');

var _promise2 = _interopRequireDefault(_promise);

var _requestPromise = require('request-promise');

var _requestPromise2 = _interopRequireDefault(_requestPromise);

var _cheerio = require('cheerio');

var _cheerio2 = _interopRequireDefault(_cheerio);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BASE_URL = 'http://www.definebabe.com';

exports.default = function (id, slug) {
  var OPTIONS = {
    uri: BASE_URL + '/gallery/' + id + '/' + slug + '/',
    followAllRedirects: true,
    transform: function transform(body) {
      return _cheerio2.default.load(body);
    }
  };
  return (0, _requestPromise2.default)(OPTIONS).then(function ($) {

    var gallery = {
      id: id,
      title: $('.gallery-block h1').text(),
      images: []
    };
    $('.gallery-block .galleries-list .thumb').map(function (i, item) {
      var image = $(item).find('a').attr('href');
      gallery.images.push(image);
    });
    return _promise2.default.resolve(gallery);
  });
};
