// definebabe/dist/lib/galleries.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _promise = require('babel-runtime/core-js/promise');

var _promise2 = _interopRequireDefault(_promise);

var _requestPromise = require('request-promise');

var _requestPromise2 = _interopRequireDefault(_requestPromise);

var _cheerio = require('cheerio');

var _cheerio2 = _interopRequireDefault(_cheerio);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BASE_URL = 'http://www.definebabe.com';

exports.default = function (id, slug) {
  var OPTIONS = {
    uri: BASE_URL + '/models/' + id + '/' + slug + '/galleries',
    followAllRedirects: true,
    transform: function transform(body) {
      return _cheerio2.default.load(body);
    }
  };
  return (0, _requestPromise2.default)(OPTIONS).then(function ($) {
    var galleries = $('.galleries-list .thumb');
    var galleriesArray = [];
    galleries.map(function (i, item) {
      galleriesArray.push({
        id: $(item).find('a').attr('href').substring(9, 13),
        title: $(item).find('a').attr('title'),
        total_pictures: $(item).find('small').text().match(/\d+/g)[0],
        thumb: $(item).find('.panorama img').attr('src')
      });
    });
    return _promise2.default.resolve(galleriesArray);
  });
};
