// definebabe/dist/lib/search.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _promise = require('babel-runtime/core-js/promise');

var _promise2 = _interopRequireDefault(_promise);

var _requestPromise = require('request-promise');

var _requestPromise2 = _interopRequireDefault(_requestPromise);

var _cheerio = require('cheerio');

var _cheerio2 = _interopRequireDefault(_cheerio);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BASE_URL = 'http://www.definebabe.com';

exports.default = function (query) {
  var OPTIONS = {
    uri: BASE_URL + '/search?words=' + query,
    followAllRedirects: true,
    transform: function transform(body) {
      return _cheerio2.default.load(body);
    }
  };
  return (0, _requestPromise2.default)(OPTIONS).then(function ($) {
    return _promise2.default.resolve({
      id: $('.profile-menu a').first().attr('href').substring(8, 12),
      name: $('div.info h1').text(),
      slug: $('div.info h1').text().replace(' ', '-').toLowerCase(),
      country: $('div.info p:contains("Country") b').text(),
      state: $('div.info p:contains("State") b').text(),
      rating: $('.rating .desc span').text()
    });
  }).catch(function (err) {
    return _promise2.default.reject(err);
  });
};
